typespec-extension/src/code-model-builder.ts

```diff
@@ -1975,6 +1987,14 @@ export class CodeModelBuilder {
 
   private processObjectSchema(type: Model, name: string): ObjectSchema {
     const namespace = getNamespace(type);
+    if (this.isArm() && isArmCommonType(type) && name?.includes("Resource")) {
+      const objectSchema = this.dummyResourceSchema(type, name);
+      this.codeModel.schemas.add(objectSchema);
+  
+      // cache this now before we accidentally recurse on this type
```

Comment:
comment seems not valid?

Reply:
Better to comment here about what you are doing.

Reply:
Seems removing the deduplication logic here still works... Looking into the cause..

Reply:
We won't recurse on this dummy schema, so not needed.